### DISCUSSION TITLE
What to base the chatsystem on?

### DISCUSSION THREAD

**Discussion by TheMorpheus407** [Bootstrap-Academy/Bootstrap-Academy#28] [Ideas]
Okay, so we want to have a chat. How it will be layouted and designed is yet up for debate, but the huge problem is:
We have lots of users on discord. It's basically support, community discussions etc.
However:
- we'd need to implement a bot that manages messages to and from discord
- there are privacy concerns with discord, since it might not be GDPR compliant

Alternative:
- Our own chat system

What are your thoughts?

**Comment by cloudishBenne** [comment]
Simplest approach:
Integrate a Matrix server implementation. There are lot of bridge modules, bots and frameworks to built a custom integration. It would be also possible to integrate Jitsi in the future.

**Reply by cloudishBenne** [reply]
Regarding GDPR compliance:
The username of a forwarded message could be replaced with a random username.
On each new standalone message it could be randomized again.
Only when a message becomes a new thread (e.g. by someone replying), the random username would be kept the same there. Parallel sent standalone messages or new threads could receive a new random username.

I don't know if this makes sense at all, just a first quick idea. Maybe there is a bridge out there, which already solves this problem.

**Reply by Cameo007** [reply]
I don't know any good alternatives to Matrix. It is much simpler to host than to code an own chat system and you can use it with existing Matrix accounts. And as @cloudishBenne said, it is very customizable.

**Reply by TheMorpheus407** [reply]
Yeah, I thought about Matrix as well. The easy solution to integrate it easily, we could create a space for the community and everyone who wants to participate in chatting or messaging would need to do so via bot?
We do host a matrix server at the moment on the server, but if we use Matrix, we could as well opt for a decentralized system and offer your own server. So we could basically use the academy as front-end for Matrix, like element of fluffy. You log in with your Matrix or connect the account and get access to messaging?

**Comment by f0xcb** [comment]
Maybe is Revolt an alternative to discord.

Landingpage: https://revolt.chat/
GitHub: https://github.com/revoltchat/

**Reply by cloudishBenne** [reply]
I love this project, but don't expect it to reach a stable version in the near future. Matrix is well established and used by governments. I think this is the more sane approach.

**Comment by 7h3Pr074g0n157** [comment]
Maybe also Spacebar could be an alternative. Its FOSS and community driven too. But it is still in developement.
GitHub: https://github.com/spacebarchat/

**Reply by Cameo007** [reply]
Than, I think it's not an option yet.

**Reply by 7h3Pr074g0n157** [reply]
Yes, I think too. 
More for completion or to keep in mind for later considerations.

**Comment by Onako2** [comment]
What about XMPP? It can be very easily extended and there are many different client and server softwares to choose from

A list can be found on https://xmpp.org/software/

It is similar like Matrix just optimized for a few million users at the same time using it.

![grafik](https://github.com/Bootstrap-Academy/Bootstrap-Academy/assets/79749977/9a126106-a34c-49eb-909b-3e00593398ef)
https://xmpp.org/uses/gaming/
